clients/src/main/java/org/apache/kafka/common/requests/ShareGroupDescribeRequest.java

```diff
@@ -0,0 +1,100 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.requests;
+
+import org.apache.kafka.common.message.ShareGroupDescribeRequestData;
+import org.apache.kafka.common.message.ShareGroupDescribeResponseData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ByteBufferAccessor;
+import org.apache.kafka.common.protocol.Errors;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ShareGroupDescribeRequest extends AbstractRequest {
+
+    public static class Builder extends AbstractRequest.Builder<ShareGroupDescribeRequest> {
+
+        private final ShareGroupDescribeRequestData data;
+
+        public Builder(ShareGroupDescribeRequestData data) {
+            this(data, false);
+        }
+
+        public Builder(ShareGroupDescribeRequestData data, boolean enableUnstableLastVersion) {
+            super(ApiKeys.SHARE_GROUP_DESCRIBE, enableUnstableLastVersion);
+            this.data = data;
+        }
+
+        @Override
+        public ShareGroupDescribeRequest build(short version) {
+            return new ShareGroupDescribeRequest(data, version);
+        }
+
+        @Override
+        public String toString() {
+            return data.toString();
+        }
+    }
+
+    private final ShareGroupDescribeRequestData data;
+
+    public ShareGroupDescribeRequest(ShareGroupDescribeRequestData data, short version) {
+        super(ApiKeys.SHARE_GROUP_DESCRIBE, version);
+        this.data = data;
+    }
+
+    @Override
+    public ShareGroupDescribeResponse getErrorResponse(int throttleTimeMs, Throwable e) {
+        ShareGroupDescribeResponseData data = new ShareGroupDescribeResponseData()
+                .setThrottleTimeMs(throttleTimeMs);
+        // Set error for each group
+        short errorCode = Errors.forException(e).code();
+        this.data.groupIds().forEach(
+                groupId -> data.groups().add(
+                        new ShareGroupDescribeResponseData.DescribedGroup()
+                                .setGroupId(groupId)
+                                .setErrorCode(errorCode)
+                )
+        );
+        return new ShareGroupDescribeResponse(data);
+    }
+
+    @Override
+    public ShareGroupDescribeRequestData data() {
+        return data;
+    }
+
+    public static ShareGroupDescribeRequest parse(ByteBuffer buffer, short version) {
+        return new ShareGroupDescribeRequest(
+                new ShareGroupDescribeRequestData(new ByteBufferAccessor(buffer), version),
+                version
+        );
+    }
+
+    public static List<ShareGroupDescribeResponseData.DescribedGroup> getErrorDescribedGroupList(
```

Comment:
Generally I have received feedback of not including `get` in such getters so may be rename method to `errorDescribedGroupList`.

Reply:
This class matches the similar `ConsumerGroupDescribeRequest` so I think it's best to leave it in the name of consistency.